### DISCUSSION TITLE
Possible to filter out everything but hardlinks?

### DISCUSSION THREAD

**Discussion by poperigby** [sharkdp/fd#1639] [Q&A]
I'd like to search for just hardlinks. Is this possible with just `fd`?

**Comment by ShreeChaturvedi** [comment]
You can do it with `--exec` and `stat`:

```sh
fd -t f -x sh -c '[ "$(stat -c %h "$1")" -gt 1 ] && printf "%s\n" "$1"' _ {}
```

`stat -c %h` gives the hard link count. The test keeps files where that count is greater than 1, meaning at least one other name points to the same inode.

On macOS, swap `stat -c %h` for `stat -f %l`.

fd doesn't have a native link-count filter (there was a request in #773 but it got closed as too niche), so this is the workaround for now.
